LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -395,17 +408,20 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
                     fpending,
                     fcntDown.GetValueOrDefault(),
                     loRaADRResult,
-                    this.logger);
+                    this.logger,
+                    this.c2dPayloadSizeHistogram);
 
                 if (cloudToDeviceMessage != null)
                 {
                     if (confirmDownlinkMessageBuilderResp.DownlinkPktFwdMessage == null)
                     {
+                        this.receiveWindowMissed?.Add(1);
                         this.logger.LogInformation($"out of time for downstream message, will abandon cloud to device message id: {cloudToDeviceMessage.MessageId ?? "undefined"}");
                         _ = cloudToDeviceMessage.AbandonAsync();
                     }
                     else if (confirmDownlinkMessageBuilderResp.IsMessageTooLong)
                     {
+                        this.receiveWindowMissed?.Add(1);
```

Comment:
This is not rx windows miss. We are not sending as the datarate the device is using doesn't allow us to carry a message of this size. Could be useful to have a metric for that as I am adding a way to change Datarate for devices, someone could programmatically request a DR change when this metric here is too frequent

Reply:
And added the metric that you suggested with 5ebf65313d67aaf1591c977d2faa0cc38488cfa9